pkg/performanceprofile/controller/performanceprofile/components/profile/profile.go

```diff
@@ -71,15 +71,19 @@ func IsPhysicalRpsEnabled(profile *performancev2.PerformanceProfile) bool {
 
 // IsRpsEnabled checks if all RPS should be applied
 func IsRpsEnabled(profile *performancev2.PerformanceProfile) bool {
-	if profile.Annotations == nil {
-		return false
-	}
-	isRpsEnabled, ok := profile.Annotations[performancev2.PerformanceProfileEnableRpsAnnotation]
-	if ok && isRpsEnabled == "true" {
-		return true
+	if profile.Annotations != nil {
```

Comment:
The convention is usually align the happy path to left.

Reply:
Hmm, but here it is a protection against dereferencing nil. Are you suggesting I move the final return up and place it under profile.Annotations == nil?

Reply:
yea and then the code can be simplified 
```
if profile.Annotations == nil {
// The default behavior enables RPS for real time workloads
		profile.Spec.WorkloadHints == nil ||
		profile.Spec.WorkloadHints.RealTime == nil || *profile.Spec.WorkloadHints.RealTime
	}

isRpsEnabled, ok := profile.Annotations[performancev2.PerformanceProfileEnableRpsAnnotation]
return ok && isRpsEnabled == "true"
```

Reply:
But that changes the logic for isRpsEnabled == "somethingstupid". I can move the ifs around, that is no bother, but I would like to keep just the two values active "true" and "false" and default to realtime workload hint value otherwise.

Reply:
isRpsEnabled != "false" is less readable I think

Reply:
> But that changes the logic for isRpsEnabled == "somethingstupid". I can move the ifs around, that is no bother, but I would like to keep just the two values active "true" and "false" and default to realtime workload hint value otherwise.

If you prefer the workload hint as the last resort then let's keep the original code

Reply:
Yeah, the hint is the default when no override is present.